File: api/src/main/java/io/bunnies/baas/resources/types/Mp4MediaType.java
package io.bunnies.baas.resources.types;

public class Mp4MediaType extends ResourceMediaType {
    public static final String KEY = "MP4";

    public Mp4MediaType(String resourceID, String mediaBaseUrl) {
        super(resourceID, mediaBaseUrl);
    }

    @Override
    public String constructUrl() {
        return this.mediaBaseUrl + "mp4/" + this.resourceID + ".mp4";
    }
}

File: api/src/main/java/io/bunnies/baas/resources/types/Mp4MediaType.kt
package io.bunnies.baas.resources.types

class Mp4MediaType(resourceID: String, mediaBaseUrl: String) : ResourceMediaType(resourceID, mediaBaseUrl) {

    override fun constructUrl(): String {
        return this.mediaBaseUrl + "mp4/" + this.resourceID + ".mp4"
    }

    companion object {
        val KEY = "MP4"
    }
}
